app/src/main/res/navigation/navigation.xml

```diff
@@ -69,9 +65,7 @@
         android:label="ShoeDetailFragment" >
         <action
             android:id="@+id/action_shoeDetailFragment_to_shoeListFragment"
-            app:destination="@id/shoeListFragment"
-            app:enterAnim="@anim/slide_in_left"
-            app:exitAnim="@anim/slide_out_right"
+            app:popUpTo="@id/shoeListFragment"
```

Comment:
Here I _swapped_ a `destination` for a `popUpTo`. This is a good example of why we even need `popUpTo` as an option. When I'm on the shoe detail screen and I press the back FAB button, I'm asking to _go back_ to the previous screen (and "discard" the current screen). If my action uses:

- `destination`: a **new** `ShoeListFragment` gets added to the top of my navigation history (and the `ShoeDetailFragment` that I'm leaving **stays** in history.
- `popUpTo`: I look back through my history to find the **most recent** `ShoeListFragment`, and then go back to that point in history (everything newer in my history gets thrown out).

The best way to "see" what's in your history is to press the Android back button (or use the back swipe gesture), which will go back through history one screen at a time by default. When using `destination` here, your history would have `ShoeDetailFragment` and `ShoeListFragment`s in it that you/a user probably wouldn't want. When using `popUpTo` here, your history doesn't have those extra `Fragment`s.

I recorded some videos to show the difference in behaviors. In both videos I use the back FAB a couple times, then see how many times I need to press the Android back button to get to the instruction `Fragment`:

### Using `destination`

https://user-images.githubusercontent.com/6463980/116894843-8669f300-ac00-11eb-8da5-97386a6fe1bc.mp4

### Using `popUpTo`

https://user-images.githubusercontent.com/6463980/116894896-9386e200-ac00-11eb-8083-839c8265d2d4.mp4

So when choosing between `destination` and `popUpTo` think about whether the action you're performing is something the user will consider to be "forward" or "backward" in history!